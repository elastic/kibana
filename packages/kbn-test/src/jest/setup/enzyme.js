/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * and the Server Side Public License, v 1; you may not use this file except in
 * compliance with, at your election, the Elastic License or the Server Side
 * Public License, v 1.
 */

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
