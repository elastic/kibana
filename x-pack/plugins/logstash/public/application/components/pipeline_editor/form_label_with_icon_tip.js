/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiIconTip } from '@elastic/eui';

export function FormLabelWithIconTip({ formRowLabelText, formRowTooltipText }) {
  return (
    <div>
      <span>{formRowLabelText}</span>
      &nbsp;
      <EuiIconTip content={formRowTooltipText} size="s" type="questionInCircle" />
    </div>
  );
}

FormLabelWithIconTip.propTypes = {
  formRowLabelText: PropTypes.string.isRequired,
  formRowTooltipText: PropTypes.string.isRequired,
};
