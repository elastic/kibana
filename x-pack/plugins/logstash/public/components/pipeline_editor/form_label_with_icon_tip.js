/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiIconTip } from '@elastic/eui';

export function FormLabelWithIconTip({ formRowLabelText, formRowTooltipText }) {
  return (
    <div>
      <span>{formRowLabelText}</span>
      &nbsp;
      <EuiIconTip content={formRowTooltipText} size="s" type="questionInCircle" />
    </div>
  );
}

FormLabelWithIconTip.propTypes = {
  formRowLabelText: PropTypes.string.isRequired,
  formRowTooltipText: PropTypes.string.isRequired,
};
