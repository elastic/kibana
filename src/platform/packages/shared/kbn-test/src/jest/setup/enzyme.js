/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

/* eslint-env jest */

/**
 * This is a workaround to fix snapshot serialization of emotion classes when rendering React@18 using `import { render } from 'enzyme'`
 * With React@18 emotion uses `useInsertionEffect` to insert styles into the DOM, which enzyme `render` does not trigger because it is using ReactDomServer.renderToString.
 * With React@17 emotion fell back to sync cb, so it was working with enzyme `render`.
 * Without those styles in DOM the custom snapshot serializer is not able to replace the emotion class names.
 * This workaround ensures a fake emotion style tag is present in the DOM before rendering the component with enzyme making the snapshot serializer work.
 */
function mockEnsureEmotionStyleTag() {
  if (!document.head.querySelector('style[data-emotion]')) {
    const style = document.createElement('style');
    style.setAttribute('data-emotion', 'css');
    document.head.appendChild(style);
  }
}

jest.mock('enzyme', () => {
  const actual = jest.requireActual('enzyme');
  return {
    ...actual,
    render: (node, options) => {
      mockEnsureEmotionStyleTag();
      return actual.render(node, options);
    },
  };
});
