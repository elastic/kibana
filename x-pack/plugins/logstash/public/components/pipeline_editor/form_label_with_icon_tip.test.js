/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { FormLabelWithIconTip } from './form_label_with_icon_tip';

describe('FormLabelWithIconTip component', () => {
  let props;

  beforeEach(() => {
    props = {
      formRowLabelText: 'label text',
      formRowTooltipText: 'tooltip text',
    };
  });

  it('renders as expected', () => {
    const wrapper = shallow(<FormLabelWithIconTip {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
