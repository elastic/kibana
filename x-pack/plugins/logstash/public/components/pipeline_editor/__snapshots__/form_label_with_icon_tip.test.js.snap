// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabelWithIconTip component renders as expected 1`] = `
<div>
  <span>
    label text
  </span>
  Â 
  <EuiIconTip
    aria-label="Info"
    content="tooltip text"
    size="s"
    type="questionInCircle"
  />
</div>
`;
